Add ValidatingAdmissionPolicy warning example

cici37 opened the pull request:
<!--

 Hello!

 Remember to ADD A DESCRIPTION and delete this note before submitting
 your pull request. The description should explain what will change,
 and why.

 PLEASE title the FIRST commit appropriately, so that if you squash all
 your commits into one, the combined commit message makes sense.
 For overall help on editing and submitting pull requests, visit:
  https://kubernetes.io/docs/contribute/suggesting-improvements/

 Use the default base branch, “main”, if you're documenting existing
 features in the English localization.

 If you're working on a different localization (not English), see
 https://kubernetes.io/docs/contribute/new-content/overview/#choose-which-git-branch-to-use
 for advice.

 If you're documenting a feature that will be part of a future release, see
 https://kubernetes.io/docs/contribute/new-content/new-features/ for advice.

-->
Fix https://github.com/kubernetes/website/issues/40952

cici37 commented:
/assign @sftim 
Thank you!

T-Lakshmi commented:
Many of the checks failed as netlify build failed due to timeout. Could you try to trigger the build again with using

git commit --amend --no-edit && git push --force-with-lease

sftim commented on content/en/blog/_posts/2020-09-03-warnings/index.md:
```diff
@@ -320,6 +320,17 @@ If you want to be sure you notice deprecations as soon as possible and get a jum
+how Validations of the referenced ValidatingAdmissionPolicy are enforced. When it sets to `Warn`, a validation failure is reported to the request client
+in HTTP Warning headers, with a warning code of 299.
+
+<img alt="kubectl applying a manifest file with ValidatingAdmissionPolicy and validationAction is set to `Warn`, displaying a warning message and exiting with a non-zero exit code."
```
```suggestion
<img alt="kubectl applying a manifest file when there is an existing a ValidatingAdmissionPolicy and a validationAction set to Warn, displaying a warning message and exiting with a non-zero exit code."
```

sftim commented on content/en/blog/_posts/2020-09-03-warnings/index.md:
```diff
@@ -320,6 +320,17 @@ If you want to be sure you notice deprecations as soon as possible and get a jum
 and required to pass with a zero exit code in order for the CI job to succeed.
 
+## Validating Admission Policy
+
```
```suggestion

_Added in 2023_

```

sftim reviewed (commented):
How about these ideas?

sftim commented:
> Many of the checks failed as netlify build failed due to timeout. Could you try to trigger the build again with using
> 
> git commit --amend --no-edit && git push --force-with-lease

Even better: rebase against _main_ (see https://github.com/kubernetes/website/issues/35780#issuecomment-1548286534)

cici37 commented:
All comments are addressed. PTAL when you have time @sftim , Thank you :)

tengqm commented on content/en/blog/_posts/2020-09-03-warnings/index.md:
```diff
@@ -320,6 +320,19 @@ If you want to be sure you notice deprecations as soon as possible and get a jum
 
+## Validating Admission Policy
+
+_Added in 2023_
```
Shall we add these as more formal documents?

sftim replied:
We do have https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/ already.

cici37 replied:
As Tim mentioned, we have a separate place to document the feature and here is the place we maintain warnings I guess :)

sftim commented on content/en/blog/_posts/2020-09-03-warnings/index.md:
```diff
@@ -320,6 +320,19 @@ If you want to be sure you notice deprecations as soon as possible and get a jum
+how Validations of the referenced ValidatingAdmissionPolicy are enforced. When it sets to `Warn`, a validation failure is reported to the request client
+in HTTP Warning headers, with a warning code of 299.
+
+<img alt="kubectl applying a manifest file when there is an existing a ValidatingAdmissionPolicy and a validationAction set to Warn, displaying a warning message and exiting with a non-zero exit code."
+src="kubectl-warnings-vap.png"
+style="width:637px;max-width:100%;">
```
This isn't rendering - see https://deploy-preview-41332--kubernetes-io-main-staging.netlify.app/blog/2020/09/03/warnings/

Please fix that and check that the [local preview](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#preview-locally) looks OK.

cici37 replied:
H @sftim , I was inspecting in the preview page and looks like the png file wasn't been built together hence it cannot find the file. Is there any recommendation on this? Thank you!

cici37 replied:
@sftim friendly ping to see if it is expected behavior(not build png file for preview) or if I did something wrong :)

sftim replied:
BTW sorry about the wait - I have a fairly long backlog of PRs etc where people have mentioned me, and I tend to use an oldest-first approach to find these

sftim commented on content/en/blog/_posts/2020-09-03-warnings/index.md:
```diff
@@ -320,6 +320,19 @@ If you want to be sure you notice deprecations as soon as possible and get a jum
+how Validations of the referenced ValidatingAdmissionPolicy are enforced. When it sets to `Warn`, a validation failure is reported to the request client
+in HTTP Warning headers, with a warning code of 299.
+
+<img alt="kubectl applying a manifest file when there is an existing a ValidatingAdmissionPolicy and a validationAction set to Warn, displaying a warning message and exiting with a non-zero exit code."
+     src="kubectl-warnings-vap.png"
+     style="width:637px;max-width:100%;">
```
```suggestion
<img alt="kubectl applying a manifest file when there is an existing a ValidatingAdmissionPolicy and a validationAction set to Warn, displaying a warning message and exiting with a non-zero exit code."
     src="kubectl-warning-vap.png"
     style="width:637px;max-width:100%;">
```

Try this. Or, rename the image.

cici37 replied:
Thank you so much for catching this! I have updated. :)

sftim reviewed (commented):
Thanks. Please fix the typo and also - because this PR predates the v1.28 release - rebase against _main_.

nate-double-u reviewed (commented):
Hi @cici37, there are a couple updates suggested for this edit. Are these changes still relevant? Would you be able to make the update and rebase if so?

cici37 commented:
> Hi @cici37, there are a couple updates suggested for this edit. Are these changes still relevant? Would you be able to make the update and rebase if so?

Thanks for the ping. I have updated and addressed all comments and squashed the commits. It should be ready to merge. :)

nate-double-u reviewed (commented):
Thanks for the updates @cici37!

@liggitt I'd like to ping you too, given you wrote the original article.

/lgtm

sftim commented:
@liggitt comments: 

> if you wanted to help people make a connection to the new thing, a single line + link at the end after a footer clearly separate from the blog pointing people at the actual docs for a new thing seems like it would be fine as well, rather than new paragraphs and screenshots

I think the screenshots sell it to me. It's too big a change if we have screenshots. @cici37 would you be willing to write a new blog article about updates to warnings since the original thing, and then revise the original article to direct readers there to read on?

You could also add a single line linking to docs (not a blog article) about the documented example. Your choice.

divya-mohan0209 commented:
/remove-lifecycle stale

divya-mohan0209 commented:
@cici37 : Would you be willing to make a separate blog PR per the suggestion by @sftim ?

divya-mohan0209 commented:
Hi @cici37 , thanks for the PR. However, since it has been ~2 months without a response, I'll be closing this one off. We encourage you to submit a separate PR for the blog instead of reopening the PR, in case you'd like to continue working on this particular subject. 

/close
